tt_eager/tt_dnn/op_library/backward/backward_ops.cpp

```diff
@@ -574,10 +574,13 @@ std::vector<Tensor> relu_bw(const Tensor& grad, const Tensor& input, const Memor
 
 std::vector<Tensor> _atan2_bw(const Tensor& grad, const Tensor& input, const Tensor& other, const MemoryConfig& output_mem_config) {
     std::vector<Tensor> grad_tensor;
+    float t_nan  = std::nanf("");
     Tensor recip_mul = mul(grad, recip(square(hypot(input,other))),std::nullopt, output_mem_config);
     Tensor grad_a = mul(other, recip_mul, std::nullopt, output_mem_config);
+    grad_a = where(logical_and(eqz(input, output_mem_config), eqz(other, output_mem_config)), t_nan, grad_a, output_mem_config);
     grad_tensor.emplace_back(grad_a);
     Tensor grad_b = mul(neg(input), recip_mul, std::nullopt, output_mem_config);
+    grad_b = where(logical_and(eqz(input, output_mem_config), eqz(other, output_mem_config)), t_nan, grad_b, output_mem_config);
```

Comment:
Seems like `logical_and(eqz(input, output_mem_config), eqz(other, output_mem_config))` computation is common for both grad_a and grad_b? Shouldn't we just compute once and reuse?

We can also deallocate `recip_mul` at an earlier point instead of leaving it in memory.
So something like:
```
Tensor grad_a = mul(other, recip_mul, std::nullopt, output_mem_config);
Tensor grad_b = mul(neg(input), recip_mul, std::nullopt, output_mem_config);
recip_mul.deallocate();
Tensor tmp = logical_and(eqz(input, output_mem_config), eqz(other, output_mem_config));
grad_a = where(tmp, t_nan, grad_a, output_mem_config);
grad_b = where(tmp, t_nan, grad_b, output_mem_config);
```